dev_apps/music-nga/js/endpoint.js

```diff
@@ -292,7 +299,17 @@ function getArtist(filePath) {
     var artist = song.metadata.artist;
 
     return new Promise((resolve) => {
-      Database.enumerateAll('metadata.artist', artist, 'next', songs => resolve(songs));
+      Database.enumerateAll('metadata.artist', artist, 'next', (songs) => {
+        songs.sort((a, b) => {
+          var albumSort = a.metadata.album < b.metadata.album ?
+            -1 : (a.metadata.album > b.metadata.album ?  1 : 0);
+
+          return albumSort || (a.metadata.discnum - b.metadata.discnum) ||
+                              (a.metadata.tracknum - b.metadata.tracknum);
+        });
```

Comment:
Ditto here. This should go in `js/db.js`. Also, you're reusing the album sorting code from above. We should just make them separate functions that get composed as necessary.